### PR TITLE
refactor: add back auto build and auto save abilities in `diff` command

### PR DESCRIPTION
**Pull request by fengxsong**
```bash
Examples:
  sealos diff labring/kubernetes:v1.25 labring/kubernetes:v1.26
  sealos diff -o table oci-archive:/path/of/older.tar oci-archive:/path/of/newer.tar
  sealos diff --patch --save -o patch.tar -t labring/kubernetes:patch-from-125-126 labring/kubernetes:v1.25 labring/kubernetes:v1.26

Flags:
      --diff-type string   type of object to diff, available options are [container, image, all] (default "image")
  -h, --help               help for diff
  -o, --output string      change the output format, available options are [json, table] (default "json")
      --patch              if enabled then create a named archive file contains changes that we interested
      --save               if enabled and --tag flag is specified, diff will save the built image into a oci-archive file with the name of --output
  -t, --tag string         tag name, diff will auto build a new image with the generated changes if flag specified and --patch=true
```

### CODE REVIEW

File: pkg/buildah/diff.go [hunk 1]
```diff
@@ -100,15 +102,16 @@ func changesToTable(out io.Writer, diffs []archive.Change) error {
...
+	save            string
 }
 
 func (o *patchOption) RegisterFlags(fs *pflag.FlagSet) {
-	fs.BoolVar(&o.createPatchFile, "create-patch", false, "If enabled then create a named archive file contains changes that we interested")
-	fs.BoolVar(&o.build, "build", false, "Auto build a new image with the generated changes")
-	fs.BoolVar(&o.save, "save", false, "Auto save the built image into a oci-archive file")
-	bailOnError(markFlagsHidden(fs, "create-patch", "build", "save"), "")
+	fs.BoolVar(&o.createPatchFile, "create-patch", false, "if enabled then create a named archive file contains changes that we interested")
+	fs.StringVar(&o.build, "build", "", "tag name, diff will auto build a new image with the generated changes if flag specified"+
```
Comment (fanux) [inline pkg/buildah/diff.go:111]: Why tag name call it `build`? Provide an example about how to use those flags.

Comment (fengxsong) [reply pkg/buildah/diff.go:111]: > Why tag name call it `build`? Provide an example about how to use those flags.

What would you recommend from a user experience perspective?

**Comment by bxy4543** [comment]
Can you add cobra.command.example for the diff command .

**Comment by fengxsong** [comment]
> Can you add cobra.command.example for the diff command .

done

**Review by cubxxw — APPROVED** [review]

**Pull request merged**